src/ng-select/ng-select.component.ts

```diff
@@ -335,15 +335,18 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
     }
 
     selectTag() {
-        let tag = {};
+        let tag;
         if (this.addTag instanceof Function) {
-            tag = this.addTag(this.filterValue)
+            tag = this.addTag(this.filterValue);
         } else {
-            tag[this.bindLabel] = this.filterValue;
+            tag = { [this.bindLabel]: this.filterValue };
         }
 
-        const item = this.itemsList.addItem(tag);
-        this.select(item);
+        if (tag instanceof Promise) {
+            tag.then(newTag => this.select(this.itemsList.addItem(newTag)));
```

Comment:
You also need to catch in case of reject otherwise you will get `Uncaught (in promise) error`

Reply:
`catch` should be done on client side since component only cares about valid resolve.